stewart_intro/simple_nn.py

```diff
@@ -0,0 +1,130 @@
+from stewart_intro.generate_training_data import normalize_data
+import numpy as np
+import pandas as pd
+from itertools import product
+import torch
+from torch.autograd import Variable
+import torch.nn.functional as F
+
+
+n_hidden_nodes = 250
+batch_size = 20
+n_epochs = 1
+n_time_steps = 1
+dt = 0.125  # 3 hours in days
+training_rate = .0001
+
+
+def get_batches(data):
+    time_values = data.isel(time=slice(0, len(data.time) - 1)).time.values
+    points = np.array(
+        [row for row in product(data.x.values, data.y.values, time_values)])
+    np.random.shuffle(points)
+    points = pd.DataFrame(points, columns=['x', 'y', 'time'])
+    n_batches = int(len(points) / batch_size)
+    batches = np.array_split(points, n_batches)
+    return batches
+
+
+def get_formatted_training_data_from_batch(data, batch):
+    batch_values = [
+        data.sel(x=row.x, y=row.y, time=row.time)
+        for _, row in batch.iterrows()
+    ]
+    x_data = np.array([
+        np.concatenate(
+            [
+                cell.QT.values,
+                cell.SLI.values,
+                [cell.LHF.values],
+                [cell.SHF.values],
+                [cell.SOLIN.values]
+            ]
+        ) for cell in batch_values
+    ])
+    fqts_fslis = [
+        np.array(
+            [np.concatenate([cell.FQT.values, cell.FSLI.values])
+             for cell in batch_values])
+    ]
+    for i in range(1, n_time_steps + 1):
+        batch_values = [
+            data.sel(x=row.x, y=row.y, time=row.time + (dt * i))
+            for _, row in batch.iterrows()
+        ]
+        if i == n_time_steps:
+            target = np.array([
+                np.concatenate([cell.QT.values, cell.SLI.values])
+                for cell in batch_values
+            ])
+        if i != n_time_steps:
+            fqts_fslis.append(np.array([
+                np.concatenate([cell.FQT.values, cell.FSLI.values])
+                for cell in batch_values
+            ]))
+    return (
+        Variable(torch.from_numpy(x_data), requires_grad=True),
+        Variable(torch.from_numpy(target), requires_grad=True),
+        Variable(torch.from_numpy(np.array(fqts_fslis)), requires_grad=True)
+    )
+
+
+def initialize_weights(data):
```

Comment:
It's totally cool that you went for the functional interface, but for deeper networks with more parameters I have really liked using pytorch's `Module` class: https://pytorch.org/docs/stable/nn.html#torch.nn.Module.
It makes it really easy to keep track of parameters, save models to pickle files, etc. If you go that route, you would make `initialize_weights` and `predict` methods of an object inheriting from `torch.nn.Module`, and declare `w1` and `w2` in the `__init__` method.

Reply:
Yes, the pytorch Module class definitely seems like the "right" way to do this -- i'll transition to that moving forward.

Reply:
ok sounds. good. let's see if your current code works before  you do too much refactoring.